source/common/http/http1/codec_impl.cc

```diff
@@ -863,6 +866,76 @@ void ConnectionImpl::onResetStreamBase(StreamResetReason reason) {
   onResetStream(reason);
 }
 
+void ConnectionImpl::dumpState(std::ostream& os, int indent_level) const {
+  const char* spaces = spacesForLevel(indent_level);
+  // Dump all bool to provide context
+  os << spaces << "Http1::ConnectionImpl " << this << DUMP_MEMBER(dispatching_)
+     << DUMP_MEMBER(dispatching_slice_already_drained_) << DUMP_MEMBER(reset_stream_called_)
+     << DUMP_MEMBER(handling_upgrade_) << DUMP_MEMBER(deferred_end_stream_headers_)
+     << DUMP_MEMBER(strict_1xx_and_204_headers_) << DUMP_MEMBER(processing_trailers_)
+     << DUMP_MEMBER(buffered_body_.length());
+
+  // Dump header parsing state, and any progress on other headers.
+  os << DUMP_MEMBER(header_parsing_state_);
+  if (header_parsing_state_ == HeaderParsingState::Field) {
+    os << DUMP_MEMBER(current_header_field_, current_header_field_.getStringView());
+  } else if (header_parsing_state_ == HeaderParsingState::Value) {
+    os << DUMP_MEMBER(current_header_field_, current_header_field_.getStringView());
+    os << DUMP_MEMBER(current_header_value_, current_header_value_.getStringView());
+  }
+
+  // Dump Child
+  os << '\n';
+  dumpAdditionalState(os, indent_level);
+
+  // Dump the first slice of the dispatching buffer if not drained.
+  if (current_dispatching_buffer_ == nullptr || dispatching_slice_already_drained_) {
+    // Buffer is either null or already drained (in the body).
+    // Use the macro for consistent formatting.
+    os << DUMP_NULLABLE_MEMBER(current_dispatching_buffer_, "drained");
+    return;
+  } else {
+    auto front_slice = [](Buffer::Instance* instance) {
+      auto slice = instance->frontSlice();
+      return absl::string_view(static_cast<const char*>(slice.mem_), slice.len_);
+    }(current_dispatching_buffer_);
```

Comment:
The previous 4 lines are hard to follow due to the inline lambda use and excessive use of 'auto'.

Consider instead:
```
absl::string_view front_slice = [](RawSlice slice) {
  return absl::string_view(static_cast<const char*>(slice.mem_), slice.len_);
}(current_dispatching_buffer_->frontSlice());
```

or

```
auto to_str_view = [](RawSlice slice) {
  return absl::string_view(static_cast<const char*>(slice.mem_), slice.len_);
};
absl::string_view front_slice = to_str_view(current_dispatching_buffer_->frontSlice());
```

Reply:
went with the former.